src/data-types/real.js

```diff
@@ -17,20 +17,28 @@ module.exports = {
   writeParameterData: function(buffer, parameter) {
     if (parameter.value != null) {
       buffer.writeUInt8(4);
-      buffer.writeFloatLE(parseFloat(parameter.value));
+      buffer.writeFloatLE(parameter.value);
     } else {
       buffer.writeUInt8(0);
     }
   },
 
-  validate: function(value) {
-    if (value == null) {
+  validate(value) {
+    if (value === undefined || value === null) {
       return null;
     }
-    value = parseFloat(value);
-    if (isNaN(value)) {
-      return new TypeError('Invalid number.');
+
+    let numberValue;
+    if (typeof value === 'number') {
+      numberValue = value;
+    } else {
+      numberValue = parseFloat(value);
     }
-    return value;
+
+    if (!Number.isFinite(numberValue) || (typeof value === 'string' && value !== numberValue.toString()) || numberValue !== Math.fround(numberValue)) {
+      return new TypeError(`The given value could not be converted to ${this.name}`);
```

Comment:
Wouldn't using Math.fround() be a breaking change? 🤔 I'm afraid that most of the data passed would fail at this point, and hinder the usage of `real` type.

Reply:
Yeah, this one's a bit harder to decide.

The loss of precision is a huge issue tho, because most users will not know that `real` is limited and won't save the exact value they're trying to send, which can lead to huge issues because the data in the database can no longer be trusted. That's why I think raising an error here is better than before and fixes a bug. What do you think? 😄

Reply:
I agree to raising error here ✌️  I was just wondering how much of an impact it would have on existing user(i.e. will it be a breaking change), but as you pointed, users must have already run into precision loss, so it better to have this in sooner rather than later.